plugins/flytekit-pydantic/flytekitplugins/pydantic/schema.py

```diff
@@ -0,0 +1,59 @@
+from typing import Type
+
+import pydantic
+from google.protobuf import json_format, struct_pb2
+
+from flytekit import FlyteContext
+from flytekit.core import type_engine
+from flytekit.models import literals, types
+
+"""
+Serializes & deserializes the pydantic basemodels
+"""
+
+
+class BaseModelTransformer(type_engine.TypeTransformer[pydantic.BaseModel]):
+    _TYPE_INFO = types.LiteralType(simple=types.SimpleType.STRUCT)
+
+    def __init__(self):
+        """Construct pydantic.BaseModelTransformer."""
+        super().__init__(name="basemodel-transform", t=pydantic.BaseModel)
+
+    def get_literal_type(self, t: Type[pydantic.BaseModel]) -> types.LiteralType:
+        return types.LiteralType(simple=types.SimpleType.STRUCT)
+
+    def to_literal(
+        self,
+        ctx: FlyteContext,
+        python_val: pydantic.BaseModel,
+        python_type: Type[pydantic.BaseModel],
+        expected: types.LiteralType,
+    ) -> literals.Literal:
+        """This method is used to convert from given python type object pydantic ``pydantic.BaseModel`` to the Literal representation."""
+
+        s = struct_pb2.Struct()
+        schema = python_val.schema_json()
+        data = python_val.json()
+        s.update({"schema": schema, "data": data})
```

Comment:
are we able to inject `flytekitplugins.pydantic.pydantic_flyteobject_config` to `python_val` somehow here, so we don't need that config when we declare the pydantic model.
```
class TrainConfig(BaseModel, **flytekitplugins.pydantic.pydantic_flyteobject_config):
   ...
```

Reply:
I like that. Applied the change in the latest commit!